tests/cbor/test_optional_fields.cpp

```diff
@@ -0,0 +1,62 @@
+#include <iostream>
+#include <rfl.hpp>
+#include <rfl/cbor.hpp>
+#include <string>
+#include <vector>
+
+#include "write_and_read.hpp"
+
+namespace test_optional_fields {
+
+struct Person {
+  rfl::Rename<"firstName", std::string> first_name;
+  rfl::Rename<"lastName", std::string> last_name = "Simpson";
+  rfl::Rename<"children", std::optional<std::vector<Person>>> children;
+
+  bool operator==(const Person& other) const {
+    return first_name.get() == other.first_name.get() &&
+           last_name.get() == other.last_name.get() &&
+           children.get() == other.children.get();
+  }
```

Comment:
Suggestion: Use the [C++20 default comparison operator](https://en.cppreference.com/w/cpp/language/default_comparisons.html):
```suggestion
  bool operator==(const Person& other) const = default;
```

Although now that I'm thinking about it, perhaps this was done because the default comparison operator is not able to use the `const Type& operator()()` that is [defined](https://github.com/getml/reflect-cpp/blob/18feb020997f61048d3141f98041cf8c964f20be/include/rfl/Rename.hpp#L72) for the `rfl::Rename<Type>` member types... That would be unfortunate.

If that's the case, it would be helpful to briefly document that fact here.

Reply:
> Although now that I'm thinking about it, perhaps this was done because the default comparison operator is not able to use the `const Type& operator()()` that is [defined](https://github.com/getml/reflect-cpp/blob/18feb020997f61048d3141f98041cf8c964f20be/include/rfl/Rename.hpp#L72) for the `rfl::Rename<Type>` member types... That would be unfortunate.

Yep, exactly